sdk/tables/Azure.Data.Tables/samples/Sample4QueryEntities.md

```diff
@@ -96,3 +96,31 @@ foreach (Page<TableEntity> page in queryResultsMaxPerPage.AsPages())
     }
 }
 ```
+
+In more advanced scenarios, it may be necessary to store the continuation token returned from the service so your code controls exactly when the next pages is fetched.
+Below is a simple example that illustrates how the token can be fetched and applied to paginated results.
+
+```C# Snippet:TablesSample4QueryPagination
+string continuationToken = null;
+bool hadResults = true;
+while (hadResults)
+{
+    Page<TableEntity> page = tableClient
+        .Query<TableEntity>()
+        .AsPages(continuationToken, pageSizeHint: 10)
+        .First(); // Note: Since the pageSizeHint only limits the number of results in a single page, we explicitly only enumerate the first page.
```

Comment:
What happens if there are no records in the table?   Should we use `FirstOrDefault` here?

Reply:
Good catch